polygon/heimdall/heimdall.go

```diff
@@ -347,6 +347,37 @@ func (h *heimdall) FetchLatestSpan(ctx context.Context) (*Span, error) {
 	return h.client.FetchLatestSpan(ctx)
 }
 
+func (h *heimdall) FetchLatestSpans(ctx context.Context, count uint) ([]*Span, error) {
+	if count == 0 {
+		return nil, errors.New("can't fetch 0 latest spans")
+	}
+
+	span, err := h.FetchLatestSpan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	latestSpans := make([]*Span, 0, count)
+	latestSpans = append(latestSpans, span)
+	count--
+
+	for count > 0 {
+		prevSpanRawId := span.RawId()
+		if prevSpanRawId == 0 {
+			return latestSpans, nil
```

Comment:
should call libcommon.SliceReverse on this before returning so that they are ordered by ID (also in service.go)

Reply:
done